resqpy/surface/_surface.py

```diff
@@ -918,6 +918,38 @@ def axial_edge_crossings(self, axis, value = 0.0):
 
         return ep
 
+    def resampled_surface(self, title = None):
+        """Creates a new triangulated set which is a resampled version of the current triangulated set. Each existing triangle in the tset is divided equally into 4 new triangles.
+           
+        arguments:
+            title (str): a new title for the output triangulated set, if None the title will have the same title as the input triangulated set
+        
+        returns:
+            resqpy.surface.Surface object, with extra_metadata ('resampled from surface': <uuid>), where uuid is the origin surface uuid
+        """
+        rt, rp = self.triangles_and_points()
+        edge1 = np.mean(rp[rt[:]][:, ::2, :], axis = 1)
+        edge2 = np.mean(rp[rt[:]][:, 1:, :], axis = 1)
+        edge3 = np.mean(rp[rt[:]][:, :2, :], axis = 1)
+        allpoints = np.concatenate((rp, edge1, edge2, edge3), axis = 0)
+        count1 = len(rp)
+        count2 = count1 + len(edge1)
+        count3 = count2 + len(edge2)
+        tris = []
+        for i in range(len(rt)):
+            tris.extend([[rt[i][0], count1 + i, count3 + i], [rt[i][1], count2 + i, count3 + i],
+                         [rt[i][2], count1 + i, count2 + i], [count1 + i, count2 + i, count3 + i]])
+
+        if title is None:
+            title = self.citation_title
+        resampled = rqs.Surface(self.model,
+                                title = title,
+                                crs_uuid = self.crs_uuid,
+                                extra_metadata = {'resampled from surface': str(self.uuid)})
+        resampled.set_from_triangles_and_points(np.array(tris), allpoints)
```

Comment:
Sorry Emma but I don't like this implementation. If I am following it correctly, we will end up with duplicated mid-edge points where an edge is common to two of the original triangles.

Also, this will re-triangulate, which is one approach. My feeling is that it might be better to derive the resampled triangles directly from the original ones, thus preserving tears for example.